diff --git a/core/lib/Drupal/Core/Field/WidgetBase.php b/core/lib/Drupal/Core/Field/WidgetBase.php
index 3d0db4ce44..d6e8497c98 100644
--- a/core/lib/Drupal/Core/Field/WidgetBase.php
+++ b/core/lib/Drupal/Core/Field/WidgetBase.php
@@ -437,7 +437,7 @@ public function flagErrors(FieldItemListInterface $items, ConstraintViolationLis
           foreach ($delta_violations as $violation) {
             // @todo: Pass $violation->arrayPropertyPath as property path.
             $error_element = $this->errorElement($delta_element, $violation, $form, $form_state);
-            if ($error_element !== FALSE) {
+            if ($error_element !== FALSE && !is_null($error_element)) {
               $form_state->setError($error_element, $violation->getMessage());
             }
           }
